lib/v2/screens/profile_screens/profile/components/logout_dialog.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:provider/provider.dart';
+import 'package:seeds/features/backup/backup_service.dart';
+import 'package:seeds/v2/blocs/authentication/viewmodels/bloc.dart';
+import 'package:seeds/design/app_theme.dart';
+import 'package:seeds/i18n/profile.i18n.dart';
+import 'package:seeds/v2/components/custom_dialog.dart';
+import 'package:seeds/v2/components/flat_button_long.dart';
+import 'package:seeds/v2/constants/app_colors.dart';
+import 'package:seeds/v2/screens/profile_screens/profile/interactor/viewmodels/bloc.dart';
+
+class LogoutDialog extends StatelessWidget {
+  const LogoutDialog({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final backupService = Provider.of<BackupService>(context);
+    return BlocBuilder<ProfileBloc, ProfileState>(
+      builder: (context, state) {
+        return CustomDialog(
+          icon: SvgPicture.asset("assets/images/profile/logout_icon.svg"),
+          children: [
+            Text('Logout'.i18n, style: Theme.of(context).textTheme.button1Black),
+            const SizedBox(height: 30.0),
+            Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 8.0),
+              child: Column(
+                children: [
+                  Text(
+                    'Save private key in secure place - to be able to restore access to your wallet later'.i18n,
+                    textAlign: TextAlign.center,
+                    style: Theme.of(context).textTheme.subtitle2Darkgreen1L,
+                  ),
+                  const SizedBox(height: 36.0),
+                  FlatButtonLong(
+                    title: 'Save private key'.i18n,
+                    onPressed: () {
+                      BlocProvider.of<ProfileBloc>(context).add(const ShowLogoutButton());
+                      backupService.backup();
+                    },
```

Comment:
I'm not sure if I should keep using this backup service or just run here what this method does is just 2 instructions.

```
  void backup() {
    Share.share(settingsStorage.privateKey!);
    settingsStorage.savePrivateKeyBackedUp(true);
  }
```

Reply:
run it here.